### PR TITLE
Trace: add client_info support to clients.

### PR DESCRIPTION
**Pull request by tseaver**
Toward #7825.

### CODE REVIEW

**Comment by tswast** [comment]
Linter failing (needs to be blackened).

Coverage failing

```
coverage report --show-missing --fail-under=100
Name                                Stmts   Miss Branch BrPart  Cover   Missing
-------------------------------------------------------------------------------
google/cloud/trace/__init__.py         12      0      0      0   100%
google/cloud/trace/_gapic.py           55      0     16      0   100%
google/cloud/trace/client.py           19      0      2      0   100%
google/cloud/trace/v1/__init__.py       3      0      0      0   100%
google/cloud/trace/v1/_gapic.py        38      2      0      0    95%   164-165
google/cloud/trace/v1/client.py        32      0     12      0   100%
google/cloud/trace_v1/__init__.py       8      0      0      0   100%
google/cloud/trace_v1/types.py         19      0      8      0   100%
google/cloud/trace_v2/__init__.py       8      0      0      0   100%
google/cloud/trace_v2/types.py         23      0      8      0   100%
-------------------------------------------------------------------------------
TOTAL                                 217      2     46      0    99%
```

https://source.cloud.google.com/results/invocations/97c40a9f-2a86-4639-b98f-11a5b146a23f/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Ftrace/log

File: trace/google/cloud/trace/client.py [hunk 1]
```diff
@@ -29,6 +34,11 @@ class Client(ClientWithProject):
         credentials (Optional[:class:`~google.auth.credentials.Credentials`]):
             The OAuth2 Credentials to use for this client. If not passed,
             falls back to the default inferred from the environment.
+        client_info (google.api_core.client_info.ClientInfo):
```
Comment (tswast) [inline trace/google/cloud/trace/client.py]: Needs gapic version of client info.

File: trace/tests/unit/v1/test__gapic_v1.py [hunk 2]
```diff
@@ -20,34 +20,30 @@
...
+
+        return _TraceAPI
+
     def _make_one(self, gapic_client=None, handwritten_client=None):
         from google.cloud.trace_v1.gapic import trace_service_client
 
-        channel = grpc_helpers.ChannelStub()
         if gapic_client is None:
-            gapic_client = trace_service_client.TraceServiceClient(channel=channel)
+            gapic_client = mock.create_autospec(trace_service_client.TraceServiceClient)
```
Comment (tswast) [inline trace/tests/unit/v1/test__gapic_v1.py]: I assume you're using a mock instead of channel stub to simplify the tests? Seems reasonable to me, if so.

Comment (tseaver) [reply trace/tests/unit/v1/test__gapic_v1.py]: Yes, as well as the fact that passing in a channel is deprecated.

File: trace/google/cloud/trace/client.py (hunk 1 above)
Comment (tseaver) [reply trace/google/cloud/trace/client.py]: 087b484

**Pull request merged**